python-checks/src/main/java/org/sonar/python/checks/hotspots/GraphQLIntrospectionCheck.java

```diff
@@ -0,0 +1,131 @@
+    "flask_graphql.GraphQLView.as_view",
+    "graphql_server.flask.GraphQLView.as_view");
+
+  private static final Set<String> SENSITIVE_VALIDATION_RULE_FQNS = Set.of(
+    "graphene.validation.DisableIntrospection",
+    "graphql.validation.NoSchemaIntrospectionCustomRule");
+
+  private static final String MESSAGE = "Disable introspection on this GraphQL server endpoint.";
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, GraphQLIntrospectionCheck::checkGraphQLIntrospection);
+  }
+
+  private static void checkGraphQLIntrospection(SubscriptionContext ctx) {
+    CallExpression callExpression = (CallExpression) ctx.syntaxNode();
+    Optional.ofNullable(callExpression.calleeSymbol())
+      .map(Symbol::fullyQualifiedName)
+      .filter(GRAPHQL_VIEWS_FQNS::contains)
+      .filter(fqn -> hasSensitiveMiddlewares(callExpression.arguments()) ||
+        hasSensitiveValidationRules(callExpression.arguments()))
+      .ifPresent(fqn -> ctx.addIssue(callExpression, MESSAGE));
+  }
+
+  private static boolean hasSensitiveMiddlewares(List<Argument> arguments) {
+    RegularArgument argument = TreeUtils.argumentByKeyword("middleware", arguments);
+    if (argument == null) {
+      return true;
+    }
+    return extractArgumentValues(argument)
+      .map(values -> values.isEmpty() || expressionsNameContainIntrospection(values))
+      .orElse(false);
+  }
+
+  private static boolean hasSensitiveValidationRules(List<Argument> arguments) {
+    RegularArgument argument = TreeUtils.argumentByKeyword("validation_rules", arguments);
+    if (argument == null) {
+      return true;
+    }
+    return extractArgumentValues(argument)
+      .map(values -> values.isEmpty() ||
+        expressionsNameContainIntrospection(values) ||
+        expressionsContainsSensitiveRuleFQN(values))
+      .orElse(false);
+  }
+
+  private static Optional<List<Expression>> extractArgumentValues(RegularArgument argument) {
+    return Optional.of(argument)
+      .map(RegularArgument::expression)
+      .flatMap(GraphQLIntrospectionCheck::expressionsFromListOrTuple);
+  }
+
+  private static Optional<List<Expression>> expressionsFromListOrTuple(Expression expression) {
+    return TreeUtils.toOptionalInstanceOf(ListLiteral.class, expression)
+      .map(ListLiteral::elements)
+      .map(ExpressionList::expressions)
+      .or(() -> TreeUtils.toOptionalInstanceOf(Tuple.class, expression)
+        .map(Tuple::elements));
+  }
+
+  private static boolean expressionsNameContainIntrospection(List<Expression> expressions) {
+    return expressions.stream()
+      .map(GraphQLIntrospectionCheck::nameFromIndentifierOrCallExpression)
+      .filter(Optional::isPresent)
+      .map(Optional::get)
+      .map(String::toUpperCase)
+      .anyMatch(name -> name.contains("INTROSPECTION"));
+  }
+
+  private static boolean expressionsContainsSensitiveRuleFQN(List<Expression> expressions) {
```

Comment:
Here then this should probably be something like `expressionsContainsSafeRuleFQN`